packages/core/js-peer/src/index.ts

```diff
@@ -0,0 +1,222 @@
+
+async function fetchResource(packageName: string, assetPath: string, version: string) {
+    if (isNode) {
+        const require = createRequire(import.meta.url);
+        
+        const file = await new Promise<Buffer>((resolve, reject) => {
+            // Cannot use 'fs/promises' with current vite config. This module is not polyfilled by default.
+            const workerFilePath = require.resolve(packageName + assetPath);
+            fs.readFile(workerFilePath, (err, data) => {
+                if (err) {
+                    reject(err);
+                    return;
+                }
+                resolve(data);
+            });
+        });
+        return new Response(file, {
+            headers: {
+                'Content-type':
+                    assetPath.endsWith('.wasm')
+                        ? 'application/wasm'
+                        : assetPath.endsWith('.js')
+                            ? 'application/javascript'
+                            : 'application/text'
+            }
+        });
+    } else {
+        // Will work everywhere fetch is supported.
+        return fetch(new globalThis.URL(`${packageName}@${version}${assetPath}`, CDN_ROOT));
+    }
+}
+
+const fetchWorkerCode = () => fetchResource('@fluencelabs/marine-worker', '', WORKER_VERSION).then(res => res.text());
+const fetchMarineJsWasm = () => fetchResource('@fluencelabs/marine-js', '/dist/marine-js.wasm', MARINE_VERSION).then(res => res.arrayBuffer());
+const fetchAvmWasm = () => fetchResource('@fluencelabs/avm', '/dist/avm.wasm', AVM_VERSION).then(res => res.arrayBuffer());
+
+const bufferToSharedArrayBuffer = (buffer: Buffer): SharedArrayBuffer => {
+    const sab = new SharedArrayBuffer(buffer.length);
+    const tmp = new Uint8Array(sab);
+    tmp.set(buffer, 0);
+    return sab;
+};
+
+export const createClient = async (relay: RelayOptions, config: ClientConfig): Promise<IFluenceClient> => {
+    const workerCode = await fetchWorkerCode();
+    const workerLoader = BlobWorker.fromText(workerCode);
+
+    const marineJsWasm = bufferToSharedArrayBuffer(Buffer.from(await fetchMarineJsWasm()));
+    const avmWasm = bufferToSharedArrayBuffer(Buffer.from(await fetchAvmWasm()));
+
+    const marine = new MarineBackgroundRunner({
+        getValue() {
+            return workerLoader;
+        }, start(): Promise<void> {
+            return Promise.resolve(undefined);
+        }, stop(): Promise<void> {
+            return Promise.resolve(undefined);
+        }
+    }, {
+        getValue() {
+            return marineJsWasm;
+        }, start(): Promise<void> {
+            return Promise.resolve(undefined);
+        }, stop(): Promise<void> {
+            return Promise.resolve(undefined);
+        }
+    }, {
+        getValue() {
+            return avmWasm;
+        }, start(): Promise<void> {
+            return Promise.resolve(undefined);
+        }, stop(): Promise<void> {
+            return Promise.resolve(undefined);
+        }
```

Comment:
I would suggest rethinking this part because seems like there is no need for this structures at all since in all of these cases start and stop do nothing and getValue is always the simple return so it can be removed as well

Reply:
I understand this is used in ephemeral networks, but I hope it can be dealt with there so we can remove it from the main implementation. Please let me know if it makes sence

Reply:
I don't want to touch this code till i work in the current task. To change code like this we need to come up with some architecture decision how we initialize dependencies. For example we can use DI or something else. Refactoring it now means we need to refactor other places like this for consistency and because they depend on each other